### DISCUSSION TITLE
V2 ghost shifting

### DISCUSSION THREAD

**Discussion by NX915** [doudar/SmartSpin2k#493] [Q&A]
Hi all! I'm facing an issue with the V2 hardware that I'm hoping to get help with. The gist of the problem is the shifters when plugged in will constantly command down shifts even when none of the button are pressed. This issue came to light when I noticed grinding noise when the unit was left plugged in but idling. It also affects any workout as the ghost shift commands causes Zwift to modify the erg difficulty continuously.

I have narrowed the issue to likely some component on the pcb. I have measured the continuity with the actual shifters and there is no short between the three contact areas on the male aux connector. The wiring harness and the female connector also tested as expected. The main culprit seem to be related to the down shift voltage pin, since I see shifting command issued whenever I contact the voltage pin for down shifting without actually bridging to ground and while I'm electrically isolated from the metal prob. In comparison, the up shift required me to hold the prob with my bare hand to trigger without bridging to ground, and that is also the same normal behavior with a separate v2 pcb that does not exhibit the same issue. I measured the two shifting pins (D18 and D19) to ground, the up shift D19 is 3.6v but the down shift D18 is only 2.3v.

The pcb I'm using is one I purchased off of eMadman. It have an extra capacitor between power and ground pin, though I think my issue is not isolated as the symptoms seems to be similar to some other discussion such as [this one](https://github.com/doudar/SmartSpin2k/discussions/473). I have resorted to disconnecting the shifter wiring harness to make the ghost shift stop, and I've been making do without the shifters. But it is inconvenient for certain workouts and ideally I would like to fix the ghost shift. Much thanks for any help and advice on this issue.

**Comment by eMadman** [comment]
As discussed on Etsy chat - sounds like this might be the 3.5mm jack.  If it's not glued into place, it's an easy swapout.  i have some presoldered with the JST connector crimped to the cable.  I can send it out for the cost of shipping.  Alternatively, you can get the jack from Amazon and solder it yourself with the existing cable:
https://www.amazon.com/gp/product/B07K55YG5X/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

**Reply by NX915** [reply]
Hi eMadman, thanks for the reply. I got your pcb in person last winter so I'm not the person you're referring to on etsy. I did test all the components including shifters, male plug, female plug, wiring harness for shorts between power and ground and found no issues. The main culprit seems to be the downshift power gpio pin D18 constantly hovering at 2.3v, which from my very limit understanding should be at 3.3v. Another strange reading I got is the D19 pin on the same board hover at around 3.6v. I tested another board I built myself and the two pins does stay around 3.3v, but that board has bluetooth connectivity issue and was the reason I purchased your's. Appreciate any insights into this strange issue.

**Reply by eMadman** [reply]
It's possible the ESP32 itself is the problem.  The devkits used for V2 were always hit or miss.  I wouldn't be surprised if it's a PCB problem.    I'll be away on holiday for the coming week but I could set you up with a V3 PCB and wiring harness at cost+shipping.  We've got quite a bit of mileage on these and they're quite a bit more stable and consistent in day to day operation.  You'll have to re-print the main case, but the motor, gear, and knob cup will transfer over without issue.

**Reply by NX915** [reply]
That would be much appreciated. I remember you were sharing the frustration with sourcing reliable esp32 last year so I'm glad to see v3 got rid of that frustrating variable. I'll reach out to you via email to take you up on the v3 board.
